public/static/app/perfdiscovery/PerfDiscoveryDataService.js

```diff
@@ -139,56 +154,58 @@ mciModule.factory('PerfDiscoveryDataService', function (
   //   },
   //   {...}
   // ]
-  function onProcessData(data) {
-    var now = {},
-      baseline = {},
-      history = {}
-
-    _.each(data, function (d) {
-      // Skip empty items
-      if (d == null) return
-
-      // TODO add group validation instead of individual
-      // Process current (revision) data
-      if (d.current && d.current.data) {
-        // Copy storageEngine to the context
-        d.ctx.storageEngine = d.current.data.storageEngine || '(none)'
-
-        _.each(d.current.data.results, function (result) {
-          processItem(result, now, d.ctx)
-        })
-      }
-
-      // Process baseline data
-      if (d.baseline && d.baseline.data) {
-        _.each(d.baseline.data.results, function (result) {
-          processItem(result, baseline, d.ctx)
-        })
-      }
+  function onProcessDataFn(metricName) {
```

Comment:
[nit] I think I still find the name a bit confusing. It is a function so the ```Fn``` suffix confuses rather than clarifies(to me anyway).

Can I suggest a more explicit name e.g. ```processDataFnFactory``` 

The follow code would also work for me:

```
function onProcessData(data, metricName) {
// doesn't return a function, just does the processing.
}
```

And called with:

```
return promise.then(data =>onProcessData(data, metricName))
```

There's not a huge difference between either (probably just where the closure for metricName is scoped).

Reply:
Ah this is much better. Fixed